### DISCUSSION TITLE
Open-Graphics: Scenery Window Additions.

### DISCUSSION THREAD

**Discussion by Sheltur** [OpenRCT2/OpenRCT2#15890] [Feature requests]
This is multiple smaller features suggested for the Scenery Window for Open Graphics and the new save-file format.

1. Separator Bars for Object types: Basically Sorting objects into their respective types, Fences, Walls, Small, Medium and Large objects with the ability to hide each sub category by clicking on a small arrow next to the bar labeling and separating them. This allows you to easily and quickly find the type of object you are looking for at a glance.
2. Search Tab: A new search tab next to the Miscellaneous Tab. This tab works as expected, a search bar exists at the top of this tab, typing in: The name of the object, Tags associated with, such as color, material, theme, or type of object (Walls, Fences etc.) This will make finding objects extremely easy and with the increased object count of the new save file format, I believe a feature like this will become increasingly more necessary as time goes on.
3. Speaking of tags, Tags: if we begin tagging each object now, I believe this will make our lives a lot easier in the future, so we don't have to do something like this later on. Tagging system, tags will include various identifiers about scenery objects including: Theme, Recolorablity, Color if not recolorable, Material such as concrete, wood and so on, As well as object type (Walls, Large and small objects etc.) Clicking on a tag does an automatic search for the item and will direct the window to the aforementioned search tab. _[potential for user added tags?]_
4. Preset Colors: Recolorable objects are no longer yellow and red when no color is selected. Instead, we will let the creator of the object will decide the default color chosen. This will let trees both be their natural color when first opening the scenery window, and also give the users the option to recolor them for let's say… a fall setting for instance. Next to the color selection a clear color button will be added, Maybe just a simple X which will clear the selected color allowing for the default colors to be chosen. If an object does not have a default color, it will simply be viewed as having the normal red and yellow colors as it is now.
5. Scenery window resizing: pretty simple, I believe letting the user scale the window to how they please should be added. Excess scenery tabs can either be moved down to a second row below the main one. Or as a second option we can have an arrow that lets the user scroll along the scenery tabs, such as in a web browser.

I believe that these features will improve the experience of using the scenery window for OpenRCT2. I personally believe they don't go too far out of the way of the “feel of, or experience of Rollercoaster Tycoon”. And with the new save file format ballooning the amount of new objects and scenery tabs that can be added, sorting through these items will be a quite the task. And these new features are designed to remedy that.

**Comment by ocalhoun6** [comment]
These all sound pretty good, except for number 4...

Doesn't the scenery window already default to the most recently used color scheme? Any new scenery placed will -- by default -- be the same colors as the previous scenery you placed. You can also use the scenery eyedropper tool to select any already-placed scenery and the default scenery colors will change to match the one you just selected.

(Unless you just started the game, of course, then RCT2's defaults are loaded. Maybe that could be improved by using the NSF to make the previous color selection persistent through saving/loading the scenario. Don't really feel like that's necessary, though.)

**Reply by Sheltur** [reply]
This is more of a way to show players how an object's supposed to look in its original form. And or show how it's intended to be used. Such as with foliage, and that is where the “clear color selection” button will come in handy. This is one of the less “useful additions”, but I believe it would be useful for replacing objects that weren't previously recolorable and visualizing how it's linked to the original asset. I actually also believe this feature is in RCT3 as well. Giving each object preset colors before choosing them yourself.

While in the “no color selected state” each piece will have its own “color” (if available) it won't be linked to one another until a color scheme is selected by the player.

That's my reasoning anyway.
Out of what I had suggested here, it's the one that I think is the least likely to make it in and the least needed but still a useful addition for future creators.

**Comment by karst** [comment]
> Excess scenery tabs can either be moved down to a second row below the main one.

This is how OpenLoco deals with excess tabs. I made a mockup on how that would look like and I think that'd be a very pleasant addition.

![image](https://user-images.githubusercontent.com/1516893/185010438-99b3aabe-c66f-4c01-87c6-7b8e97b7cfff.png)

Additionally I think it'd be preferable to have the object authors listed either in the middle on the bottom (in White) or on the complete right and move price to the center.
